### ISSUE TITLE
PyArrow Integration

### ISSUE THREAD

**Issue by paddyhoran** [InvestmentSystems/static-frame#38]
I listened to the TalkPython episode today that covered this library and I realized that this is the library I was looking for about a year ago.  I have implemented parts of this library several times trying to address the pain points it tries to solve, so thank you!

Is the [Apache Arrow](https://arrow.apache.org/) project on your radar at all?  It is very much aligned with the goals of this library and would provide several other features for free.

**Comment by flexatone** [comment]
Many thanks for your comments!

I have been looking at Arrow for some time. While our focus at this time has been on interface design, as we move forward to consider broader performance issues, there might be interesting options in working with PyArrow as an alternative to NumPy as a data back end. In the nearer term, supporting to / from conversion from Arrow and Parquet is desirable and practical.

**Comment by flexatone** [comment]
I have implemented basic support for creating and reading Arrow Tables, and writing to and reading from Parquet binary files. Presently this supports basic types and `IndexHierarchy` on both axis; additional work will be needed for some specialized dtypes and object arrays. This will be released as part of v0.5.0. 

It appears in some cases NumPy arrays can be created from Arrow Array views, however this is not supported presently on the ChunkedArray, which is the column representation given from the Array Table. (https://arrow.apache.org/docs/python/numpy.html)

I will close this issue now; please open new issues if you are looking for any other specific Arrow features. 

```python
>>> from static_frame import Frame

>>> f1 = Frame.from_dict(dict(a=(1,2), b=(False, True)), index=('x', 'y'))
>>> f1.to_arrow()
pyarrow.Table
__index0__: string
a: int64
b: bool

>>> Frame.from_arrow(f1.to_arrow(), index_depth=1)
<Frame>
<Index>  a       b      <<U1>
<Index>
x        1       False
y        2       True
<object> <int64> <bool>

>>> f1.to_parquet('/tmp/test.parquet')
>>> f2 = Frame.from_parquet('/tmp/test.parquet', index_depth=1)
>>> f2
<Frame>
<Index>  a       b      <<U1>
<Index>
x        1       False
y        2       True
<object> <int64> <bool>
